src/NuGet.Services.AzureSearch/SearchService/AzureSearchQuery.cs

```diff
@@ -0,0 +1,161 @@
+            _fieldScopedValues = new Dictionary<string, List<string>>();
+        }
+
+        public void AddNonFieldScopedValues(IEnumerable<string> values)
+        {
+            _nonFieldScopedValues.AddRange(values);
+        }
+
+        public void AddFieldScopedValues(string fieldName, IEnumerable<string> values)
+        {
+            if (!_fieldScopedValues.ContainsKey(fieldName))
+            {
+                _fieldScopedValues[fieldName] = new List<string>();
+            }
+
+            _fieldScopedValues[fieldName].AddRange(values);
+        }
+
+        public override string ToString()
+        {
+            var result = new StringBuilder();
+
+            foreach (var fieldScopedTerm in _fieldScopedValues)
+            {
+                // Separate groups of field-scoped terms are AND'd together.
+                if (result.Length > 0)
+                {
+                    result.Append(" AND ");
+                }
+
+                // If there are both non-field-scoped terms and field-scoped terms, only the field-scoped terms
+                // are required.
+                if (_nonFieldScopedValues.Any())
+                {
+                    result.Append('+');
+                }
+
+                result.Append(fieldScopedTerm.Key);
+                result.Append(':');
+
+                if (fieldScopedTerm.Value.Count == 1)
+                {
+                    AppendEscapedString(result, fieldScopedTerm.Value[0]);
+                }
+                else
+                {
+                    result.Append('(');
+                    AppendEscapedValues(result, fieldScopedTerm.Value);
+                    result.Append(')');
+                }
+            }
+
+            if (_nonFieldScopedValues.Any())
+            {
+                var wrapWithParenthesis = false;
+                if (result.Length > 0)
+                {
+                    result.Append(' ');
+                }
+                else
+                {
+                    // If there are no field-scoped terms, at least one non-scoped term is required.
+                    result.Append('+');
+                    result.Append('(');
+                    wrapWithParenthesis = true;
+                }
+
+                AppendEscapedValues(result, _nonFieldScopedValues);
+
+                if (wrapWithParenthesis)
+                {
+                    result.Append(')');
+                }
+            }
+
+            return result.ToString();
+        }
+
+        private static void AppendEscapedValues(StringBuilder result, IReadOnlyList<string> values)
+        {
+            for (var i = 0; i < values.Count; i++)
+            {
+                if (i > 0)
+                {
+                    result.Append(' ');
+                }
+
+                AppendEscapedString(result, values[i]);
+            }
+        }
+
+        private static void AppendEscapedString(StringBuilder result, string input)
+        {
```

Comment:
What the legacy service's behavior on surrogate pairs? This is kind of a nasty rabbit hole be we should know what the story is.

Reply:
😈 you can punt this till later but they shouldn't like cause an exception or give a backdoor

Reply:
Moved to separate issue: https://github.com/NuGet/NuGetGallery/issues/6939